tmt/steps/report/reportportal.py

```diff
@@ -1,152 +1,221 @@
+            assert launch_uuid is not None
+            self.info("uuid", launch_uuid, "yellow", shift=1)
+
+            # For each test
+            for result in self.step.plan.execute.results():
+                test = [test for test in self.step.plan.discover.tests()
+                        if test.serialnumber == result.serialnumber][0]
+                item_attributes = attributes.copy()
+                if test.contact:
+                    item_attributes.append({"key": "contact", "value": test.contact[0]})
+                env_vars = [{'key': k, 'value': v} for k, v in test.environment.items()
+                            if not k.startswith("TMT_") or tmt_vars]
+
+                #  Create a test item
+                self.info("test", result.name, color="cyan")
+                response = session.post(url=f"{url}/item",
+                                        headers=headers,
+                                        json={"name": result.name,
+                                              "description": test.summary,
+                                              "attributes": item_attributes,
+                                              "parameters": env_vars,
+                                              "codeRef": test.web_link() or None,
+                                              "launchUuid": launch_uuid,
+                                              "type": "step",
+                                              "testCaseId": test.id or None,
+                                              "startTime": timestamp()})
+                self.handle_response(response)
+                item_uuid = yaml_to_dict(response.text).get("id")
+                assert item_uuid is not None
+                self.info("uuid", item_uuid, "yellow", shift=1)
+
+                # For each log
+                for log_path in result.log:
+                    try:
+                        log = self.step.plan.execute.read(log_path)
+                    except tmt.utils.FileError:
+                        continue
+
+                    level = "INFO" if log_path == result.log[0] else "TRACE"
+                    status = self.TMT_TO_RP_RESULT_STATUS[result.result]
+
+                    # Upload logs
+                    response = session.post(url=f"{url}/log/entry",
+                                            headers=headers,
+                                            json={"message": log,
+                                                  "itemUuid": item_uuid,
+                                                  "launchUuid": launch_uuid,
+                                                  "level": level,
+                                                  "time": timestamp()})
+                    self.handle_response(response)
+
+                    # Write out failures
+                    if log_path == result.log[0] and status == "FAILED":
+                        message = result.failures(self.step.plan.execute.read(result.log[0]))
```

Comment:
There's probably no need to call `read()` again: if `log_path == result.log[0]`, and we're processing the first log, then `log` from line 177 should already hold the content of the said log. `result.failures(log)` should give the same outcome.

Reply:
Thanks, I might have overlooked this one.